### DISCUSSION TITLE
Restrict access to data which has been created by user only

### DISCUSSION THREAD

**Discussion by ghost** [pocketbase/pocketbase#1749] [Q&A] [answered]
Hey I'm playing and learn more about pocketbase + react native.  Is there a way to fetch datas which has been added by a specific user only with getFullList? I want to secure any read access which has been created by the user. 

To fetch datas I'm using 
const records = await pb.collection('todolist').getFullList

In this case I get all datas from any different users. 

What I expect:
Get all datas from the authenticated user only

**Comment by ganigeorgiev** [comment, marked as answer]
This could be achieved using the Collection API rules. You can find some information and examples in https://pocketbase.io/docs/api-rules-and-filters/.

If your `todolist` collection has a nonempty `user` `relation` field, then you can set a simple List and View rule like:

```js
user = @request.auth.id
```

**Reply by ghost** [reply]
Thank you so much for the quick response. This was the solution for me "user = @request.auth.id". 

I had passed wrong api rules in the 'todolist' collection which has a relation with user
